### PR TITLE
Create check-refund endpoint for multiple selected accounts

### PR DESCRIPTION
**Pull request by mrdpopov**
## Purpose

Create check-refund endpoint for multiple selected accounts

## Approach

* Tests improvements: Common methods have been moved to a new parent  class.
* Tests improvements: Tests for bulk pay, waive and transfer have tests have been segregated into a separate class to address erroneous implementation done previously: tests for single action check were executed multiple times due to parametrization in tests. 
* Bulk check refund tests have been added in the image of single check refund tests.

Jira: https://issues.folio.org/browse/MODFEE-111

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -910,6 +919,16 @@
...
+    {
+      "permissionName":"accounts.bulk.check-transfer.post",
+      "displayName":"accounts transfer check",
+      "description":"Checks if transfer is allowed"
+    },
+    {
+      "permissionName":"accounts.bulk.check-waive.post",
+      "displayName":"accounts waive check",
+      "description":"Checks if waive is allowed"
+    },
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:931]: Do we need these permissions?

File: src/main/java/org/folio/rest/impl/AccountsAPI.java [hunk 2]
```diff
@@ -2,18 +2,13 @@
 
 import static io.vertx.core.Future.succeededFuture;
 
-import io.vertx.core.AsyncResult;
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/AccountsAPI.java:5]: Is import ordering the only thing that has changed in this file?

File: src/test/java/org/folio/rest/impl/accountactionchecks/AccountsActionChecksAPITests.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+  @Before
+  public void setUp() {
+    firstAccount = createAccount();
+    secondAccount = createAccount();
+    accountsCheckPayClient = buildAccountCheckPayClient(firstAccount.getId());
+    accountsCheckWaiveClient = buildAccountCheckWaiveClient(firstAccount.getId());
+    accountsCheckTransferClient = buildAccountCheckTransferClient(firstAccount.getId());
+  }
+
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/accountactionchecks/AccountsActionChecksAPITests.java:28]: Blank line

Comment (mrdpopov) [reply src/test/java/org/folio/rest/impl/accountactionchecks/AccountsActionChecksAPITests.java:28]: done.

File: src/main/java/org/folio/rest/impl/AccountsAPI.java (hunk 2 above)
Comment (mrdpopov) [reply src/main/java/org/folio/rest/impl/AccountsAPI.java:5]: done.

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (mrdpopov) [reply descriptors/ModuleDescriptor-template.json:931]: done.

File: src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkCheckPayWaiveTransferAPITests.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import org.folio.rest.domain.Action;
+import org.folio.rest.impl.accountactionchecks.AccountsActionChecksAPITestsBase;
+import org.folio.rest.utils.ResourceClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class AccountsActionBulkCheckPayWaiveTransferAPITests extends AccountsActionChecksAPITestsBase {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkCheckPayWaiveTransferAPITests.java:19]: Indentation?

File: src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkRefundAPITests.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+import org.apache.http.HttpStatus;
+import org.folio.rest.impl.accountactionchecks.AccountsActionChecksAPITestsBase;
+import org.folio.rest.jaxrs.model.Feefineaction;
+import org.folio.rest.utils.ResourceClient;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AccountsActionBulkRefundAPITests extends AccountsActionChecksAPITestsBase {
+
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkRefundAPITests.java:17]: Blank line

**Review by alexanderkurash — APPROVED** [review]

File: src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkRefundAPITests.java (hunk 5 above)
Comment (mrdpopov) [reply src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkRefundAPITests.java:17]: done.

**Comment by mrdpopov** [comment]
Regarding those 36 Code Smells. These are all "Add at least one assertion to this test case" in tests that were moved to a new package. The assertions happen with help of RestAssured and adding was deemed as a bad idea.

File: src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkCheckPayWaiveTransferAPITests.java (hunk 4 above)
Comment (mrdpopov) [reply src/test/java/org/folio/rest/impl/accountactionchecks/bulk/AccountsActionBulkCheckPayWaiveTransferAPITests.java:19]: done.

**Pull request merged**